lib/decanter/core.rb

```diff
@@ -76,10 +76,11 @@ def required_inputs
         end  
       end
 
-      def required_input_values_present?(args={})
+      def required_input_keys_present?(args={})
         return true unless any_inputs_required?
-        required_inputs.all? do |input|
-          args.keys.include?(input)
+        compact_inputs = required_inputs.compact
```

Comment:
Could also add `.with_indifferent_access` here instead of the extra mapping `to_sym` step